### ISSUE TITLE
--scope adds unwanted packages

### ISSUE THREAD

**Issue by MrSquaare** [vercel/turborepo#978]
### What version of Turborepo are you using?

1.1.10

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Linux

### Describe the Bug

I have several packages, especially `frontend` and `ui-kit`.

`frontend` has `ui-kit` dependency in its `package.json`.

When I run some command that exists in both `frontend` and `ui-kit`, both packages are selected, even if I clearly specify `--scope=ui-kit`.

Example:
```
turbo run start --scope=ui-kit --dry-run
```
results in:
```
Packages in Scope
Name     Path     
frontend frontend 
ui-kit   ui-kit   

...
```

Also, there is no `dependsOn` for these rules in the `turbo.json`

### Expected Behavior

Having only the specified package selected

### To Reproduce

1. Having two packages
  - `a`, with a `test` script
  - `b`, depending on `a` in it's package.json's `dependencies`, with a `test` script
2. Executing `turbo run test --scope=a`

**Comment by jlarmstrongiv** [comment]
Works on mac, broken on linux

**Comment by TitusKirch** [comment]
> EDIT: can no longer reproduce 😬

I can still reproduce the behaviour under Linux.

**Comment by jaredpalmer** [comment]
This should be fixed in turbo@1.2

**Comment by MrSquaare** [comment]
@jaredpalmer I still have this issue with turbo 1.2.1
